### PR TITLE
Owner-Report can display business document links

### PR DESCRIPTION
**Pull request by christopherlam**
Partial PR. Handle `invoice-doclink-guid=` links. This PR currently sends doclink URI in URL; it should send invoice GUID, and `top-level.c` would retrieve gncInvoice from GUID. Not sure how to do this. @Bob-IT ?

Not sure whether to add doclink as a separate column, or linkify the "Invoice" in the Type column.

Also "Display/Document Links" will now clash with "Display/Links" option.

Progress so far:
image

### CODE REVIEW

**Comment by Bob-IT** [comment]
@christopherlam  change top-level.c as follows...
add #include "gncInvoice.h" at line 50 
add GncInvoice  * invoice; at line 124
and make the if statement at 185 look like....

    else if (strncmp ("invoice-doclink-guid=", location,
                      strlen ("invoice-doclink-guid=")) == 0)
    {
        if (!validate_type("invoice-doclink-guid=", location, GNC_ID_INVOICE,
                           result, &guid, &entity))
            return FALSE;

        invoice = (GncInvoice *) entity;
        gnc_doclink_open_uri (gnc_ui_get_gtk_window (GTK_WIDGET (result->parent)),
                              gncInvoiceGetDocLink (invoice));
        return TRUE;
    }

Now do your magic scheme stuff in new-owner-report.scm line 808
I fudged it by doing...
;;         (gncInvoiceGetDocLink invoice)
         (gncInvoiceReturnGUID invoice)
but I imagine you will need to do someing like the trep engine.

I would keep it as a separate column that can be added or not and maybe add 'Enable Links' like the 'Transaction Report'

**Comment by christopherlam** [comment]
Nice, works well. Now what remains is i10n and strings. We should standardise them:

In Owner Report:
Document Links = doclinks
Links = invoice->payments and payment->invoices

In Transaction report:
Enable Links = allow HTML anchors. (This could be removed... is redundant because stylesheet has option to disable links)

image
image

File: gnucash/report/reports/standard/new-owner-report.scm [hunk 1]
```diff
@@ -56,6 +56,7 @@
 (define credit-header (N_ "Credits"))
 (define debit-header (N_ "Debits"))
 (define balance-header (N_ "Balance"))
+(define doclink-header (N_ "Document Links"))
 (define linked-txns-header (N_ "Links"))
```
Comment (fellen) [inline gnucash/report/reports/standard/new-owner-report.scm:60]: What kind of link are this? If we have "Document Links", it should become "Transaction Links" or whatever it is.

Comment (christopherlam) [reply gnucash/report/reports/standard/new-owner-report.scm:60]: These are links from data file to external Uri describing files or web address.

Comment (fellen) [reply gnucash/report/reports/standard/new-owner-report.scm:60]: doclink-header (N_ "Document Links") is obvisious.
I meant linked-txns-header (N_ "Links")

Comment (jralls) [reply gnucash/report/reports/standard/new-owner-report.scm:60]: It's the transactions linked to the bill/invoice/voucher/credit note via the "post" operation.

Comment (christopherlam) [reply gnucash/report/reports/standard/new-owner-report.scm:60]: So:

Owner Report:
`Document Links` = doclinks
`Transaction Links` vs `Business Links` = invoice->payments and payment->invoices

In Transaction report:
`Enable Links` - remove?

**Pull request merged**